echopype/mask/api.py

```diff
@@ -17,6 +17,231 @@
+        If ``mask`` is a single-element and ``storage_options_mask`` is not a single dict
+    TypeError
+        If ``storage_options_mask`` is not a list of dict or a dict
+    """
+
+    if isinstance(mask, list):
+
+        # if storage_options_mask is not a list create a list of
+        # length len(mask) with elements storage_options_mask
+        if not isinstance(storage_options_mask, list):
+
+            if not isinstance(storage_options_mask, dict):
+                raise TypeError("storage_options_mask must be a list of dict or a dict!")
+
+            storage_options_mask = [storage_options_mask] * len(mask)
+        else:
+            # ensure all element of storage_options_mask are a dict
+            if not all([isinstance(elem, dict) for elem in storage_options_mask]):
+                raise TypeError("storage_options_mask must be a list of dict or a dict!")
+
+        for mask_ind in range(len(mask)):
+
+            # validate the mask type or path (if it is provided)
+            mask_val, file_type = validate_source_ds_da(
+                mask[mask_ind], storage_options_mask[mask_ind]
+            )
+
+            # replace mask element path with its corresponding DataArray
+            if isinstance(mask_val, str):
+                # open up DataArray using mask path
+                mask[mask_ind] = xr.open_dataarray(
+                    mask_val, engine=file_type, chunks="auto", **storage_options_mask[mask_ind]
+                )
+
+    else:
+
+        if not isinstance(storage_options_mask, dict):
+            raise RuntimeError(
+                "The provided input storage_options_mask should be a single "
+                "dict because mask is a single value!"
+            )
+
+        # validate the mask type or path (if it is provided)
+        mask, file_type = validate_source_ds_da(mask, storage_options_mask)
+
+        if isinstance(mask, str):
+            # open up DataArray using mask path
+            mask = xr.open_dataarray(mask, engine=file_type, chunks="auto", **storage_options_mask)
+
+    return mask
+
+
+def _check_var_name_fill_value(
+    source_ds: xr.Dataset, var_name: str, fill_value: Union[int, float, np.ndarray, xr.DataArray]
+) -> None:
+    """
+    Ensures that the inputs ``var_name`` and ``fill_value`` for the function
+    ``apply_mask`` were appropriately provided.
+
+    Parameters
+    ----------
+    source_ds: xr.Dataset
+        A Dataset that contains the variable ``var_name``
+    var_name: str
+        The variable name in ``source_ds`` that the mask should be applied to
+    fill_value: int or float or np.ndarray or xr.DataArray
+        Specifies the value(s) at false indices
+
+    Raises
+    ------
+    TypeError
+        If ``var_name`` or ``fill_value`` are not an accepted type
+    RuntimeError
+        If the Dataset ``source_ds`` does not contain ``var_name``
+    RuntimeError
+        If ``fill_value`` is an array and not the same shape as ``var_name``
```

Comment:
Please do overarching changes of `RuntimeError` to `ValueError` in this PR.

We'll worry about existing ones in other recent PRs later.